app/controllers/videos_controller.rb

```diff
@@ -63,11 +60,14 @@ def destroy
   private
     # Use callbacks to share common setup or constraints between actions.
     def set_video
-      @video = Video.find(params[:id])
+      @video = Video.find_by(id: params[:id])
+      if @video.nil?
+        flash[:alert] = "video not avalible"
+        redirect_to "/videos#index"
```

Comment:
@moizansari22 Kindly use named route.
https://guides.rubyonrails.org/routing.html#controller-namespaces-and-routing